cpp/modmesh/transform/fourier.hpp

```diff
@@ -63,22 +67,37 @@ class FourierTransform
     FourierTransform & operator=(FourierTransform && other) = delete;
 
     template <template <typename> class T1, typename T2>
-    static void fft(SimpleArray<T1<T2>> const & in, SimpleArray<T1<T2>> & out)
+    static void fft(SimpleArray<T1<T2>> const & in, SimpleArray<T1<T2>> & out, std::string const && backend)
     {
         const size_t N = in.size();
 
-        if ((N & (N - 1)) == 0)
+        if (backend == "cpu")
```

Comment:
String comparison can only be done during runtime, but the comparison should be available during compile-time.